earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
+
+    # Install extra debian packages we will need to support plugins.
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        graphviz
+
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    RUN poetry config installer.max-workers 10
+
+    # Copy our dependencies.
+    COPY pyproject.toml .
```

Comment:
The `COPY` command can take multiple arguments. I'm not sure if Earthly optimizes for this or not, but technically you've created three layers here when only one is needed (move all of the arguments into a single `COPY` command). If any one of these files change, there's no speed increase by separating them because copy operations are cheap.